load-test-framework/src/main/java/com/google/pubsub/clients/common/MetricsHandler.java

```diff
@@ -0,0 +1,222 @@
+          String tempZoneId = EntityUtils.toString(zoneIdResponse.getEntity());
+          if (tempZoneId.lastIndexOf("/") >= 0) {
+            zoneId = tempZoneId.substring(tempZoneId.lastIndexOf("/") + 1);
+          } else {
+            zoneId = tempZoneId;
+          }
+          HttpGet instanceIdRequest =
+              new HttpGet("http://metadata.google.internal/computeMetadata/v1/instance/id");
+          instanceIdRequest.setHeader("Metadata-Flavor", "Google");
+          HttpResponse instanceIdResponse = httpClient.execute(instanceIdRequest);
+          instanceId = EntityUtils.toString(instanceIdResponse.getEntity());
+        } catch (IOException e) {
+          log.info(
+              "Unable to connect to metadata server, assuming not on GCE, setting "
+                  + "defaults for instance and zone.");
+          instanceId = "local";
+          zoneId = "us-east1-b";  // Must use a valid cloud zone even if running local.
+        }
+
+        monitoredResource.setLabels(ImmutableMap.of(
+            "project_id", project,
+            "instance_id", instanceId,
+            "zone", zoneId
+        ));
+        createMetrics();
+        executor.scheduleAtFixedRate(this::reportMetrics, 30, 30, TimeUnit.SECONDS);
+      } catch (IOException e) {
+        log.error("Unable to initialize MetricsHandler, trying again.", e);
+        executor.execute(this::initialize);
+      } catch (GeneralSecurityException e) {
+        log.error("Unable to initialize MetricsHandler permanently, credentials error.", e);
+      }
+    }
+  }
+
+  private void createMetrics() {
+    try {
+      MetricDescriptor metricDescriptor = new MetricDescriptor()
+          .setType("custom.googleapis.com/cloud-pubsub/loadclient/" + metricName);
+      metricDescriptor.setDisplayName(metricName.toPrettyString())
+          .setDescription(metricName.toPrettyString())
+          .setName(metricDescriptor.getType())
+          .setLabels(Collections.singletonList(new LabelDescriptor()
+              .setKey("client_type")
+              .setDescription("The type of client reporting latency.")
+              .setValueType("STRING")))
+          .setMetricKind("GAUGE")
+          .setValueType("DISTRIBUTION")
+          .setUnit("ms");
+      monitoring.projects().metricDescriptors().create("projects/" + project, metricDescriptor).execute();
+    } catch (Exception e) {
+      log.info("Metrics already exist.");
+    }
+  }
+
+  public synchronized void recordLatency(long latencyMs) {
+    distribution.recordLatency(latencyMs);
+  }
+
+  private void reportMetrics() {
+    CreateTimeSeriesRequest request;
+    synchronized (this) {
+      String now = dateFormatter.format(new Date());
+      request = new CreateTimeSeriesRequest().setTimeSeries(Collections.singletonList(
+          new TimeSeries()
+              .setMetric(new Metric()
+                  .setType("custom.googleapis.com/cloud-pubsub/loadclient/" + metricName)
```

Comment:
This can be anything that starts with custom.googleapis.com and that does not already exist in your project.